ethereumj-core/src/main/java/org/ethereum/net/server/ChannelManager.java

```diff
@@ -153,7 +153,7 @@ public void connect(Node node) {
         return ids;
     }
 
-    private void processNewPeers() {
+    private synchronized void processNewPeers() {
```

Comment:
Such things like `channel.disconnect()` under lock make me nervous :(
I.e. we are calling method which can potentially traverse many components. This is very deadlock prone approach imho